LiteCore/Database/VectorDocument.cc

```diff
@@ -563,11 +563,20 @@ namespace litecore {
                 } else if ( !winningVersion.hasVersionVector() || !losingVersion.hasVersionVector() ) {
                     // At least one side had a legacy rev tree ID, so this requires some fuss.
                     if ( localWon ) {
-                        // Convert to a version vector up front, along the lines of "server branch switch"
-                        // that we used to do when the local won in rev tree mode.
-                        mergedVersion = VersionVector::trivialMerge(toVersionVector(winningVersion),
-                                                                    toVersionVector(losingVersion));
-                        mergedRevID   = mergedVersion.asBinary();
+                        if ( !winningVersion.hasVersionVector() && !losingVersion.hasVersionVector() ) {
+                            // Both sides are legacy rev ID which makes a trivial merge not possible
```

Comment:
What I am concerned is, even though hasVersionVector() is true, it could be a fake vector as produced by toVersionVector() because of, for instance, a previous invocation of this function.